modules/components/tag-input/tag-input.ts

```diff
@@ -509,11 +509,9 @@ export class TagInputComponent extends TagInputAccessor implements OnInit, After
      * @name appendTag
      * @param tag {TagModel}
      */
-    public appendTag = (tag: TagModel): void => {
-        this.items = [
-            ...this.items,
-            this.modelAsStrings ? tag[this.identifyBy] : tag
```

Comment:
@eyalhakim here is the issue - if modelAsStrings is true, you need to append a string rather than an object